File: rust/src/models/feature_ref.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{entitlement_product_ref::EntitlementProductRef, feature_id::FeatureId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct FeatureRef {
    /// Unique key used to reference this feature in your code. Cannot be changed after creation.
    pub code: String,

    pub id: FeatureId,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub product: Option<EntitlementProductRef>,
}

impl FeatureRef {
    pub fn new(code: String, id: FeatureId, name: String) -> Self {
        Self {
            code,
            id,
            name,
            product: None,
        }
    }
}

File: browser/src/models/featureRef.ts
import {
  type EntitlementProductRef,
  EntitlementProductRefSerializer,
} from "./entitlementProductRef";
import { type FeatureId, FeatureIdSerializer } from "./featureId";

export interface FeatureRef {
  /** Unique key used to reference this feature in your code. Cannot be changed after creation. */
  code: string;

  id: FeatureId;

  name: string;

  product?: EntitlementProductRef | null;
}

export const FeatureRefSerializer = {
  _fromJsonObject(object: any): FeatureRef {
    return {
      code: object["code"],
      id: FeatureIdSerializer._fromJsonObject(object["id"]),
      name: object["name"],
      product:
        object["product"] != null
          ? EntitlementProductRefSerializer._fromJsonObject(object["product"])
          : undefined,
    };
  },

  _toJsonObject(self: FeatureRef): any {
    return {
      code: self.code,
      id: FeatureIdSerializer._toJsonObject(self.id),
      name: self.name,
      product:
        self.product != null
          ? EntitlementProductRefSerializer._toJsonObject(self.product)
          : undefined,
    };
  },
};
